src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs

```diff
@@ -931,7 +931,20 @@ protected virtual JwtSecurityToken ValidateSignature(string token, TokenValidati
                 return validatedJwt;
             }
 
-            JwtSecurityToken jwtToken = ReadJwtToken(token);
+            JwtSecurityToken jwtToken = null;
+
+            if (validationParameters.TokenReader != null)
+            {
+                var securityToken = validationParameters.TokenReader(token);
+                if (securityToken == null)
+                    throw LogHelper.LogExceptionMessage(new SecurityTokenInvalidSignatureException(LogHelper.FormatInvariant(TokenLogMessages.IDX10507, token)));
+
+                jwtToken = securityToken as JwtSecurityToken;
+                if (jwtToken == null)
+                    throw LogHelper.LogExceptionMessage(new SecurityTokenInvalidSignatureException(LogHelper.FormatInvariant(TokenLogMessages.IDX10508, typeof(JwtSecurityToken), securityToken.GetType(), token)));
+            } else
```

Comment:
In this case we should put the else on a separate line and then use { }. Even though the else has only one clause.

Reply:
Fixed it! I'll try to be more careful next time!